packages/expo-cli/src/projects.ts

```diff
@@ -50,4 +51,25 @@ async function ensureProjectExistsAsync(
   }
 }
 
-export { ensureProjectExistsAsync };
+/**
+ * Get the account and project name using a user and Expo config.
+ * This will validate if the owner field is set when using a robot account.
+ */
+function getProjectData(
+  user: User | RobotUser,
+  exp: ExpoConfig
+): Pick<ProjectData, 'accountName' | 'projectName'> {
+  if (user.kind === 'robot' && !exp.owner) {
+    throw new CommandError(
+      'ROBOT_OWNER_ERROR',
+      'The "owner" manifest property is required when using robot users.'
+    );
+  }
+
+  return {
+    accountName: exp.owner || user.username,
+    projectName: exp.slug,
+  };
```

Comment:
```suggestion
  return exp.owner || user.username;
```
I'd rename this function to `getProjectOwner(user: User | RobotUser, exp: ExpoConfig): string` and only return the project owner. Why? These "bag of fields" functions tend to grow beyond their original intended purpose: they'll be changed to return more fields and will be used to get these fields even if there's a more direct way to get them. A historical example of this:
https://github.com/expo/expo-cli/blob/cd782b4c5b580f0aaaa03229e174b7657bf26a4a/packages/xdl/src/Exp.ts#L314-L316

(Another way to view this is adhering to the "single-responsibility principle".)

Also the `exp.owner || user.username` expression is repeated in many places and it will be useful to have a focused function that can be called to do it. (That also does the check for robots.)

Reply:
Good idea, I renamed it!